tools/build_docker_image.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+set -e
+
+FIRECRACKER_SRC_DIR=$(pwd)
+FIRECRACKER_TOOLS_DIR="$FIRECRACKER_SRC_DIR/tools"
+TEMP_IMAGE_ID_LOCATION=$(mktemp)
+
+freeze_requirements_if_not_present() {
+    expected_requirements_freeze_file_path="$FIRECRACKER_TOOLS_DIR/requirements-freeze.txt"
+    if [[ ! -f "$expected_requirements_freeze_file_path" ]]; then
+       cp "$FIRECRACKER_TOOLS_DIR/requirements.txt" $expected_requirements_freeze_file_path
+    fi
+}
+
+build_docker_image() {
+    docker build --iidfile $TEMP_IMAGE_ID_LOCATION -t fcuvm/dev -f "tools/devctr/$1" . 
+    echo "Completed buidling docker image."
+}
+
+get_image_id() {
+    image_identifier=$(head -n 1 $TEMP_IMAGE_ID_LOCATION)
+    # parse image_id from sha256:image_id...xxx
+    image_id=${image_identifier##*:}
+    rm $TEMP_IMAGE_ID_LOCATION
+
+    echo $image_id
+}
+
+# Given a valid docker image id the script pulls requirements-freeze.txt from container
+# and updates/creates requirements-freeze.txt on host.
+main () {
+    if [ $# -eq 0 ]
+       then
+        echo "No docker file name provided, exiting..."
+        exit 1
+    fi
+
+    while [[ "$#" -gt 0 ]]; do
+        case $1 in
+            -f|--docker-file) docker_file_name="$2"; shift ;;
+            *) echo "Invalid parameter passed: $1"; exit 1 ;;
+        esac
+        shift
+    done
+    
+    freeze_requirements_if_not_present
+    build_docker_image $docker_file_name
+    image_id=$(get_image_id)
+    bash $FIRECRACKER_TOOLS_DIR/update_requirements.sh -i $image_id
```

Comment:
This path doesn't exist. It should be renamed to `update_pip_requirements.txt`

Reply:
thanks a lot. fixed the bug in new iteration. 

Tested by re-running the script against Dockerfile.aarch64 and Dockerfile.x86_64.